File: apps/web/src/types/mcp.ts
export interface MCPServer {
  id: string
  transport: string
  command: string
  args: string[]
  env: Record<string, string>
}

export interface MCPServerHealth {
  id: string
  ok: boolean
  tool_count: number
  latency_ms: number
  error?: string | null
  tools_sample?: string[]
}

export interface MCPRuntimeStatus {
  ok: boolean
  error?: string | null
  tool_count: number
  config_path: string
}

export interface ToolAlias {
  logical_name: string
  mcp_tool?: string | null
  server?: string | null
  description: string
  arg_map: Record<string, string>
  defaults: Record<string, unknown>
  use_mcp: boolean
  allow_in_skills?: string[]
}

export interface MCPOverview {
  runtime: MCPRuntimeStatus
  servers: MCPServer[]
  server_health: MCPServerHealth[]
  aliases: ToolAlias[]
  tools: string[]
}

export interface MCPServerCreateRequest {
  id: string
  transport?: string
  command: string
  args?: string[]
  env?: Record<string, string>
}

File: apps/api/app/schemas/mcp.py
"""MCP management API schemas."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class MCPServer(BaseModel):
    id: str
    transport: str = "stdio"
    command: str | None = None
    args: list[str] = Field(default_factory=list)
    env: dict[str, str] = Field(default_factory=dict)


class MCPServerCreateRequest(BaseModel):
    id: str = Field(..., min_length=1, max_length=64, pattern=r"^[a-zA-Z0-9_-]+$")
    transport: str = "stdio"
    command: str
    args: list[str] = Field(default_factory=list)
    env: dict[str, str] = Field(default_factory=dict)


class MCPServerHealth(BaseModel):
    id: str
    ok: bool
    tool_count: int = 0
    latency_ms: int = 0
    error: str | None = None
    tools_sample: list[str] = Field(default_factory=list)


class MCPRuntimeStatus(BaseModel):
    ok: bool
    error: str | None = None
    tool_count: int = 0
    config_path: str


class ToolAlias(BaseModel):
    logical_name: str
    mcp_tool: str | None = None
    server: str | None = None
    description: str = ""
    arg_map: dict[str, str] = Field(default_factory=dict)
    defaults: dict[str, Any] = Field(default_factory=dict)
    use_mcp: bool = True
    allow_in_skills: list[str] = Field(default_factory=list)


class MCPOverviewResponse(BaseModel):
    runtime: MCPRuntimeStatus
    servers: list[MCPServer]
    server_health: list[MCPServerHealth]
    aliases: list[ToolAlias]
    tools: list[str]


class MCPReloadResponse(BaseModel):
    ok: bool
    runtime: MCPRuntimeStatus
    message: str
